addons/mail/static/src/components/thread_viewer/thread_viewer.xml

```diff
@@ -31,6 +31,7 @@
                         hasCurrentPartnerAvatar="props.hasComposerCurrentPartnerAvatar"
                         hasSendButton="props.hasComposerSendButton"
                         hasThreadTyping="props.hasComposerThreadTyping"
+                        isCompact="!threadViewer.thread.mass_mailing"
```

Comment:
Not sure what is `isCompact`, but could use attr from composer model in composer component instead of using prop?

Reply:
if `isCompact` is true then it the composer is on one line with the send button on the right, else it looks like in chatter or mass_mailing with the buttons underneath the textearea.

I'm not sure it can be done with entity as this is computed on creation of composer so we would need to override creation of threadviewer to handle this logic, not sure if it's something we want ? or isCompact should be a computed field that depends on threadviewer and other potential linked entities (which is something that is not totally functional for the moment as we need to trigger the compute is some cases). And I feel that this is putting too much logic in composer as I'm not sure it's his duty to know if he has to be compact or not, it's more on the behalve of the parent I think.

Reply:
`isCompact` depends on where/how it is presented. If we want to move it in models, we should model each views, similarly to `mail.thread_viewer`. Right now, `mail.composer` is a sub-model of `mail.thread` and acts as the persistent shared state of all UI composers.

In short, this is too much effort to include in this PR, so let's keep logic in template right now.

Reply:
ok